dot/parachain/overseer/overseer.go

```diff
@@ -231,31 +280,28 @@ func (o *OverseerSystem) handleBlockEvents() {
 				}
 				return
 			}
-
 			o.activeLeaves[imported.Header.Hash()] = uint32(imported.Header.Number)
-			delete(o.activeLeaves, imported.Header.ParentHash)
 
 			var activeLeavesUpdate parachaintypes.ActiveLeavesUpdateSignal
 
-			supports, err := o.DoesHeadSupportsParachainConsensus(imported.Header.Hash())
-			if err != nil {
-				panic(fmt.Sprintf("checking head supports parachain: %s", err.Error()))
-			}
-
-			if supports {
+			if o.onHeadActivated(imported.Header.Hash()) {
 				activeLeavesUpdate = parachaintypes.ActiveLeavesUpdateSignal{
 					Activated: &parachaintypes.ActivatedLeaf{
 						Hash:   imported.Header.Hash(),
 						Number: uint32(imported.Header.Number),
 					},
-					Deactivated: []common.Hash{imported.Header.ParentHash},
 				}
 			}
 
+			if _, ok := o.activeLeaves[imported.Header.ParentHash]; ok {
+				activeLeavesUpdate.Deactivated = append(activeLeavesUpdate.Deactivated, imported.Header.Hash())
```

Comment:
Why is the imported blocks hash being added to `Deactivated` instead of its parent hash?

Reply:
The parent is not an active leaf any more. But the block that was just imported is added to `o.activeLeaves` in line 283 above.